src/coreclr/jit/lowerarmarch.cpp

```diff
@@ -2552,8 +2557,74 @@ void Lowering::ContainCheckNeg(GenTreeOp* neg)
 }
 
 //----------------------------------------------------------------------------------------------
-// Try converting SELECT/SELECTCC to CINC/CINCCC. Conversion is possible only if
-// both the trueVal and falseVal are integral constants and abs(trueVal - falseVal) = 1.
+// TryLowerCselToCinvOrCneg: Try converting SELECT/SELECTCC to SELECT_INV/SELECT_INVCC. Conversion is possible only if
+// one of the operands of the select node is inverted.
+//
+// Arguments:
+//     select - The select node that is now SELECT or SELECTCC
+//     cond   - The condition node that SELECT or SELECTCC uses
+//
+void Lowering::TryLowerCselToCinvOrCneg(GenTreeOp* select, GenTree* cond)
+{
+    assert(select->OperIs(GT_SELECT, GT_SELECTCC));
+
+    bool     shouldReverseCondition;
+    GenTree* invertedOrNegatedVal;
+    GenTree* nonInvertedOrNegatedVal;
+
+    GenTree*   trueVal  = select->gtOp1;
+    GenTree*   falseVal = select->gtOp2;
+    const bool isCneg   = ((trueVal->gtOper == GT_NEG) || (falseVal->gtOper == GT_NEG));
+
+    if ((trueVal->gtOper == GT_NOT) || (trueVal->gtOper == GT_NEG))
+    {
+        shouldReverseCondition  = true;
+        invertedOrNegatedVal    = trueVal->AsOp()->gtOp1;
+        nonInvertedOrNegatedVal = falseVal;
+        BlockRange().Remove(trueVal);
+    }
+    else
+    {
+        shouldReverseCondition  = false;
+        invertedOrNegatedVal    = falseVal->AsOp()->gtOp1;
+        nonInvertedOrNegatedVal = trueVal;
+        BlockRange().Remove(falseVal);
+    }
+
+    select->gtOp1 = nonInvertedOrNegatedVal;
+    select->gtOp2 = invertedOrNegatedVal;
+
+    if (select->OperIs(GT_SELECT))
+    {
+        if (shouldReverseCondition)
+        {
+            GenTree* revCond = comp->gtReverseCond(cond);
+            assert(cond == revCond); // Ensure `gtReverseCond` did not create a new node.
```

Comment:
Does anything guarantee that this will work? I don't think so. You should structure the code so that we can bail out without making any modifications in this case if `!cond->OperIsCompare()`,

Reply:
Added a bailout logic before making any changes to a node.